src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -128,21 +128,32 @@ allocation_type ocl_engine::detect_usm_allocation_type(const void* memory) const
 bool ocl_engine::check_allocatable(const layout& layout, allocation_type type) {
     OPENVINO_ASSERT(supports_allocation(type) || type == allocation_type::cl_mem, "[GPU] Unsupported allocation type: ", type);
 
-    OPENVINO_ASSERT(layout.bytes_count() <= get_device_info().max_alloc_mem_size,
+    bool above_max_device_mem_size = (layout.bytes_count() > get_device_info().max_alloc_mem_size);
+
+    // When dynamic shape upper bound makes bigger buffer, then return false.
+    if (above_max_device_mem_size && layout.is_dynamic())
```

Comment:
Please add a check of layout.has_upper_bound() too, because allocation for dynamic layout which does not have upper bound is an error, so it it is good to check.

Reply:
Updated! Thanks!